mm/kasan/kasan.c

```diff
@@ -222,188 +222,188 @@ void __asan_handle_no_return(void)
   /* Shut up compiler complaints */
 }
 
-void __asan_report_load_n_noabort(uintptr_t addr, size_t size)
+void __asan_report_load_n_noabort(FAR void *addr, size_t size)
 {
-  kasan_report(addr, size, false);
+  kasan_report((uintptr_t)addr, size, false);
```

Comment:
let's change the prototype of kasan_report and kasan_set_posion to avoid the cast